lib/chef/provisioning/convergence_strategy/precreate_chef_objects.rb

```diff
@@ -226,10 +234,43 @@ def client_rb_content(chef_server_url, node_name)
             https_proxy #{convergence_options[:bootstrap_proxy].inspect}
           EOM
         end
+        if convergence_options[:policy_name]
+          content << <<-EOM
+            # Policyfile Settings:
+            use_policyfile true
+            policy_document_native_api true
+            policy_group "#{convergence_options[:policy_group]}"
+            policy_name "#{convergence_options[:policy_name]}"
+          EOM
+        end
+
         content.gsub!(/^\s+/, "")
         content << convergence_options[:chef_config] if convergence_options[:chef_config]
         content
       end
+
+      def validate_policy_if_enabled
+        raise "Policy_name was given but no Policy_group. Both are required" if (convergence_options[:policy_name] &&
```

Comment:
`'policy_name'` instead of Policy_name, since that's the name of the actual attribute/option. Same for `'policy_group`'